docs/scos/dev/glue-api-guides/202204.0/glue-backend-api/security-and-authentication.md

```diff
@@ -0,0 +1,29 @@
+---
+title: Security and Authentication
+description: This article describes the authorization mechanism used in Spryker, as well as user scopes, database tables, and extension points.
+last_updated: Oct 24, 2022
+template: glue-api-storefront-guide-template
+---
+
+When exposing information via Spryker Glue API and integrating with third-party applications, it is essential to protect API endpoints from unauthorized access. For this purpose, Spryker provides an authorization mechanism, using which you can request users to authenticate themselves before accessing a resource. For this purpose, Spryker Glue is shipped with an implementation of the OAuth 2.0 protocol. It allows users to authenticate themselves with their username and password and receive an access token. The token can then be used to access protected resources.
+
+The authorization mechanism provided by the Glue is flexible and allows you to define which endpoints specifically require authentication. Usually, protected endpoints include customer wish lists, carts, personal data and the like. Also, you may want to protect additional areas as required by your project. In addition to endpoints, you may require authorization to use specific REST verbs. For example, a certain endpoint can allow retrieving information, but not modifying it. In this case, the GET verb can be allowed for usage without authentication, while the PUT, PATCH and DELETE verbs will require user authentication to use.
+
+## How authentication works
+
+To authenticate a user, the client must send an authentication request, containing the username and password. If the provided credentials match a valid Spryker user, the API responds with a 201 response code and a message containing an access token, also known as a bearer token. The token can then be used to access protected API resources. Schematically, the authentication and authorization scheme of Spryker REST API can be represented as follows:
```

Comment:
not completely true. It depends on grant type. Username and password are required only for the `password` grant type